docs/keps/20200619-federated-resource-status.md

```diff
@@ -0,0 +1,514 @@
+	LastTransitionTime metav1.Time `json:"lastTransitionTime" protobuf:"bytes,4,opt,name=lastTransitionTime"`
+	// The reason for the condition's last transition in CamelCase.
+	// The specific API may choose whether or not this field is considered a guaranteed API.
+	// This field may not be empty.
+	// +required
+	Reason string `json:"reason" protobuf:"bytes,5,opt,name=reason"`
+	// A human readable message indicating details about the transition.
+	// This field may be empty.
+	// +required
+	Message string `json:"message" protobuf:"bytes,6,opt,name=message"`
+}
+```
+
+Nowadays `Conditions` field of a federated resource holds the status of the federated actions (aka propagation status).
+In other words, it defines the conditions of the propagation status for a resource.
+
+```yaml
+- apiVersion: types.kubefed.io/v1beta1
+  kind: FederatedDeployment
+  metadata:
+    finalizers:
+    - kubefed.io/sync-controller
+    generation: 1
+    name: asystem
+    namespace: asystem
+    resourceVersion: "70174497"
+  spec:
+    placement:
+      clusters:
+      - name: cluster3
+      - name: cluster2
+      - name: cluster1
+    template:
+      metadata:
+        labels:
+          app: nginx
+      spec:
+        replicas: 3
+        selector:
+          matchLabels:
+            app: nginx
+        template:
+          metadata:
+            labels:
+              app: nginx
+          spec:
+            containers:
+            - image: nginx
+              name: nginx
+  status:
+    conditions:
+    - lastTransitionTime: "2020-05-25T19:47:59Z"
+      lastUpdateTime: "2020-05-25T19:47:59Z"
+      status: "True"
+      type: Propagation
+```
+
+`status.conditions` reports the latest status which defines the state of the propagation.
+Obviously it is not necessary to report all the clusters for which the propagation was completed
+successfully.
+In relation to this, this approach does not change the initial implementation where
+the status is only collected for the services and in case of an error during the propagation.
+Therefore the proposed solution updates the `status/conditions` and `clusters/conditions` to
+identify the state of a federated resource in case of failure or non-ready state.
+
+The intention in this proposal is to extend the current available `Clusters` status with an additional field `Conditions` to
+hold the status of the federated resources, e.g Ready, NotReady.
+
+The status of the federated resources should determine whether the resources satisfy
+a `Ready` condition, and otherwise report their error status.
+To do so, this property reports the status of the federated resources in their
+target clusters whenever a `ReadyCondition` is not satisfied.
+This condition would need to be identified per type or by the usage of an interface
+`IsReady` that determines this value per type of resource.
+By doing so, we ensure the `Clusters/Conditions` property shows the status of only unhealthy
+resources for the respective clusters.
```

Comment:
I feel that for each cluster, we should also include the `Propagation` condition as well. If for any reason, the propagation of the federated resource failed, we should also report that at per cluster basis.